python/hail/expr/expressions/typed_expressions.py

```diff
@@ -1657,7 +1522,8 @@ def __neg__(self):
         :class:`.NumericExpression`
             Negated number.
         """
-        return self._unary_op("-")
+
+        return expr_numeric.coerce(self)._unary_op("-")
```

Comment:
this is the only numeric op which doesn't go through the logic above, so this will convert bools to int.